flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala

```diff
@@ -24,4 +24,19 @@ package org.apache.flink.table.functions
   * User-defined functions must have a default constructor and must be instantiable during runtime.
   */
 trait UserDefinedFunction {
```

Comment:
Did you create a Java class which extends `ScalarFunction` or `TableFunction` ? I tried this in my environment but the compiling is failed.

```java
public class MyScalarFunction extends ScalarFunction {
	public int eval(int a) {
		return a + 1;
	}
}
```

```
Error:(25, 8) java: org.apache.flink.table.examples.java.MyScalarFunction不是抽象的,
 并且未覆盖org.apache.flink.table.functions.UserDefinedFunction中的抽象方法close()
```